File: 3788-maximum-unique-subarray-sum-after-deletion/maximum-unique-subarray-sum-after-deletion.py
class Solution(object):
    def maxSum(self, nums):
        if len(nums)==1:
            return nums[0]
        unq=sorted(set(nums))
        sum=0
        for n in unq:
            if(n>0):
                sum+=n
        if(sum>0):
            return sum
        return unq[-1]            

        
        

File: 3788-maximum-unique-subarray-sum-after-deletion/maximum-unique-subarray-sum-after-deletion.cpp
class Solution {
public:
    int maxSum(vector<int>& nums) {
        set<int>s;
        int n=nums.size();
        if(n==1)return nums[0];
        int max_sum=0;
        for(int i:nums){
            s.insert(i);
        }
        for(int i:s){
            if(i>0){
                max_sum+=i;
            }
        }
        if(max_sum>0){
            return max_sum;
        }
        return *s.rbegin();
    }
};